### DISCUSSION TITLE
LabVIEW FPGA command line

### DISCUSSION THREAD

**Discussion by amitwadje** [JamesMc86/G-CLI#170] [Q&A]
Hello,
Does G-CLI have functionality to compile LabVIEW FPGA using command line ?

**Comment by JamesWiresmith** [comment]
No unfortunately FPGA doesn't follow the standard build API and has it's own API.

I've tried to do something similar myself but it gets more complicated as it launches the compiles asynchronously. @joerghampel may have done something with this?

**Reply by joerghampel** [reply]
As James says, the compilation is technically completely different from simply executing a build specification and requires a lot of steps under the hood: Creating intermediary files, connecting to build engine (local tools or compile server), queuing for the next slot, keeping synced up with the parallel task of the compilation itself, collecting results...).

Indeed, our Release Automation Tools (-> https://rat.hampel-soft.com) support the programmatic compilation of FPGA bitfiles. We have not productized this feature yet, it is still in beta phase. Technically, it's working fine, but it needs more polishing before we can call it a product. We're dependent on some commercial support to get it done.
